### ISSUE TITLE
ng add @angular/pwa fails when missing "test" architect

### ISSUE THREAD

**Issue by lacolaco** [angular/angular-cli#12374]
### Bug Report or Feature Request (mark with an `x`)
```
- [x] bug report -> please search issues before submitting
- [ ] feature request
```

### Command (mark with an `x`)
```
- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [x] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Versions

```
Angular CLI: 6.2.3
Node: 8.10.0
OS: darwin x64
Angular: 6.1.8
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.8.3
@angular-devkit/build-angular     0.8.3
@angular-devkit/build-optimizer   0.8.3
@angular-devkit/build-webpack     0.8.3
@angular-devkit/core              0.8.3
@angular-devkit/schematics        0.8.3
@angular/cli                      6.2.3
@angular/pwa                      0.8.3
@ngtools/webpack                  6.2.3
@schematics/angular               0.8.3
@schematics/update                0.8.3
rxjs                              6.2.2
typescript                        2.9.2
webpack                           4.20.2
```

### Repro steps

1. Run `$ ng new some-workspace`
2. Remove `test` architect from `angular.json`
3. Run `$ ng add @angular/pwa` and it fails.

### The log given by the failure

```
> ng add @angular/pwa
Installing packages for tooling via yarn.
yarn add v1.9.4
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
success Saved lockfile.
success Saved 1 new dependency.
info Direct dependencies
└─ @angular/pwa@0.8.3
info All dependencies
└─ @angular/pwa@0.8.3
✨  Done in 5.86s.
Installed packages for tooling via yarn.
Cannot read property 'options' of undefined
```

### Desired functionality

Adding assets process should be run only for the actual available architect and If `test` architect is missing, it should be skipped.

### Mention any other details that might be useful

CLI's default `test` architect is not needed when the project uses Jest for testing. 
`test` architect is not mandatory.

**Comment by clydin** [comment]
This is currently fixed in master and will be available in 7.0 (currently in beta).

**Comment by lacolaco** [comment]
Thanks!
